subsys/bluetooth/mesh/shell/shell_scheduler_cli.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdlib.h>
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+#include <sys/math_extras.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "../model_utils.h"
+
+#include "shell_utils.h"
+
+static struct bt_mesh_model *mod;
+static struct bt_mesh_schedule_entry set_entry;
+
+static int cmd_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCHEDULER_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scheduler_cli *cli = mod->user_data;
+	uint16_t rsp;
+
+	int err = bt_mesh_scheduler_cli_get(cli, NULL, &rsp);
+
+	if (!err) {
+		shell_print(shell, "Status: %d", rsp);
+	}
+
+	return err;
+}
+
+static void schedule_print(const struct shell *shell, int err, struct bt_mesh_schedule_entry *rsp)
+{
+	if (!err) {
+		shell_print(shell,
+			    "Day: %02d month: %02d year: %02d,\n"
+			    "Hour: %d minute:%d second:%d,\n"
+			    "Weekday: %d, action: %d, trans time: %d, scene num: %d",
+			    rsp->day, (u32_count_trailing_zeros(rsp->month) + 1), rsp->year,
+			    rsp->hour, rsp->minute, rsp->second,
+			    (u32_count_trailing_zeros(rsp->day_of_week) + 1), rsp->action,
+			    model_transition_decode(rsp->transition_time), rsp->scene_number);
+	}
+}
+
+static int cmd_action_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint8_t idx = (uint8_t)strtol(argv[1], NULL, 0);
+
+	if (!mod && !shell_model_first_get(BT_MESH_MODEL_ID_SCHEDULER_CLI, &mod)) {
+		return -ENODEV;
+	}
+
+	struct bt_mesh_scheduler_cli *cli = mod->user_data;
+	struct bt_mesh_schedule_entry rsp;
+
+	int err = bt_mesh_scheduler_cli_action_get(cli, NULL, idx, &rsp);
+
+	schedule_print(shell, err, &rsp);
+	return err;
+}
+
+static int cmd_action_ctx_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	set_entry.year = (uint8_t)strtol(argv[1], NULL, 0);
+	set_entry.day = (uint8_t)strtol(argv[3], NULL, 0);
+	set_entry.hour = (uint8_t)strtol(argv[4], NULL, 0);
+	set_entry.minute = (uint8_t)strtol(argv[5], NULL, 0);
+	set_entry.second = (uint8_t)strtol(argv[6], NULL, 0);
+	set_entry.action = (enum bt_mesh_scheduler_action)strtol(argv[8], NULL, 0);
+	set_entry.scene_number = (uint16_t)strtol(argv[10], NULL, 0);
+
+	uint8_t month = (uint8_t)strtol(argv[2], NULL, 0);
+	uint8_t day_of_week = (uint8_t)strtol(argv[7], NULL, 0);
+	uint32_t transition_time = (uint32_t)strtol(argv[9], NULL, 0);
+
+	set_entry.month = BIT(month - 1);
```

Comment:
I'm not sure I got this logic. It is possible to provide a mask of days of week or\and months. It shouldn't be mandatory only a single day or month.

Reply:
yes, you are right. I did this before you pointed out the sheduler.h comment. My initial thought was that providing a bitmask via decimal is not ideal, but i see now that it wont work.